Mathlib/Algebra/Group/Defs.lean

```diff
@@ -426,6 +426,24 @@ instance (priority := 100) RightCancelSemigroup.toIsRightCancelMul (G : Type u)
 #align right_cancel_semigroup.to_is_right_cancel_mul RightCancelSemigroup.toIsRightCancelMul
 #align add_right_cancel_semigroup.to_is_right_cancel_add AddRightCancelSemigroup.toIsRightCancelAdd
 
+/-- An `AddCancelSemigroup` is an additive semigroup such that `a + b = a + c` implies `b = c`,
+and `a + c = b + c` implies `a = b`. -/
+class AddCancelSemigroup (α : Type*) extends AddLeftCancelSemigroup α, AddRightCancelSemigroup α
+
+/-- A `CancelSemigroup` is a semigroup such that `a * b = a * c` implies `b = c`, and
+`a * c = b * c` implies `a = b`. -/
+@[to_additive]
+class CancelSemigroup (α : Type*) extends LeftCancelSemigroup α, RightCancelSemigroup α
+
+attribute [to_additive existing] CancelSemigroup.toRightCancelSemigroup
+
+/-- Any cancellative semigroup has cancellative multiplication. -/
+@[to_additive toIsCancelAdd "Any cancellative semigroup has cancellative addition."]
+instance (priority := 100) CancelSemigroup.toIsCancelMul (M : Type*) [CancelSemigroup M] :
+    IsCancelMul M :=
+  { mul_left_cancel := LeftCancelSemigroup.mul_left_cancel
+    mul_right_cancel := RightCancelSemigroup.mul_right_cancel }
```

Comment:
Does this work?
```suggestion
  { }
```

(or perhaps a lone `where`)

It should be implied by the instance above.